Tests/AfluentTests/DeferredTaskTests.swift

```diff
@@ -26,4 +27,46 @@ struct AfluentTests {
             }.run()
         }
     }
+
+    @Test(.timeLimit(.milliseconds(20))) func deferredTaskCancelledWithinCancelledTask_WithExecute() async throws {
+        await #expect(throws: CancellationError.self) {
+            try await withMainSerialExecutor {
```

Comment:
Did this prove main serial executor necessary because of flakiness locally?

Reply:
I did, yes. Another option would be to use an additional single value subject to indicate the `DeferredTask` has started. Lmk if you would prefer that over the current implementation.

Reply:
Nope, what's here makes sense. Thanks!